src/config.cpp

```diff
@@ -546,17 +567,39 @@ class SimulationConfig::Parser
             auto& valueIt = it.value();
             const auto debugStr = fmt::format("report {}", it.key());
             parseMandatory(valueIt, "cells", debugStr, report.cells);
+            parseOptional<std::string>(valueIt, "sections", report.sections, "soma");
             parseMandatory(valueIt, "type", debugStr, report.type);
+            parseOptional<std::string>(valueIt, "scaling", report.scaling, "area");
+            if (report.sections == "soma") {
+                parseOptional<std::string>(valueIt, "compartments", report.compartments, "center");
+            } else {
+                parseOptional<std::string>(valueIt, "compartments", report.compartments, "all");
+            }
+            parseMandatory(valueIt, "variable_name", debugStr, report.variableName);
+            parseOptional<std::string>(valueIt, "unit", report.unit, "mV");
             parseMandatory(valueIt, "dt", debugStr, report.dt);
             parseMandatory(valueIt, "start_time", debugStr, report.startTime);
             parseMandatory(valueIt, "end_time", debugStr, report.endTime);
-            parseOptional(valueIt, "file_name", report.fileName);
-            if (report.fileName.empty())
-                report.fileName = it.key() + "_SONATA.h5";
-            else {
-                const auto extension = fs::path(report.fileName).extension().string();
-                if (extension.empty() || extension != ".h5")
-                    report.fileName += ".h5";
+            parseOptional<std::string>(valueIt,
+                                       "file_name",
+                                       report.fileName,
+                                       it.key() + "_SONATA.h5");
+            parseOptional(valueIt, "enabled", report.enabled);
+
+            checkValidField<std::string>(report.sections, {"soma", "axon", "dend", "apic", "all"});
+            checkValidField<std::string>(report.scaling, {"none", "area"});
+            checkValidField<std::string>(report.compartments, {"center", "all"});
+
+            // Validate comma separated strings
+            std::regex expr("^\\w+(\\s*,\\s*\\w+)*$");
```

Comment:
IIRC, the `^` and `$` aren't needed for `regex_match`, since it matches the whole thing.

The grouping can be non-capturing, since we don't extract results.

Using a raw string literal will also mean the double backslashes aren't needed.

Reply:
Lets close then this PR and continue the conversation in #197